### PR TITLE
Adds reconciliation gauge to hydra ingest

### PR DESCRIPTION
**Pull request by mcguffin0725**

### CODE REVIEW

File: kafka/src/main/scala/hydra/kafka/ingestors/KafkaIngestor.scala [hunk 1]
```diff
@@ -16,26 +16,46 @@
...
     case Ingest(record, ackStrategy) => transport(record, ackStrategy)
   }
+
+  override def validateRequest(request: HydraRequest): Try[HydraRequest] = {
+    val topic = request.metadataValue(HYDRA_KAFKA_TOPIC_PARAM).getOrElse("topicMissing")
+
+    HydraMetrics.incrementGauge(
+      lookupKey = ReconciliationGaugeName + s"_$topic",
+      metricName = ReconciliationMetricName,
+      tags = Seq("ingestType" -> "kafka", "topic" -> topic)
```
Comment (ivanwakeup) [inline kafka/src/main/scala/hydra/kafka/ingestors/KafkaIngestor.scala]: so we still need the ack strategy on here if we're going to do monitoring, we can really only alert on metrics with a persisted ack level

File: kafka/src/main/scala/hydra/kafka/transport/KafkaTransport.scala [hunk 2]
```diff
@@ -68,11 +72,19 @@ class KafkaTransport(producerSettings: Map[String, ProducerSettings[Any, Any]])
           "transport" -> persistenceId
         )
       )
+
+      HydraMetrics.decrementGauge(
+        KafkaIngestor.ReconciliationGaugeName + s"_$topic",
+        KafkaIngestor.ReconciliationMetricName,
+        tags = Seq("ingestType" -> "kafka", "topic" -> topic)
```
Comment (ivanwakeup) [inline kafka/src/main/scala/hydra/kafka/transport/KafkaTransport.scala]: think you mentioned the ack level isn't available here? idk if the decrement behavior works correctly if the tags aren't congruent

File: kafka/src/main/scala/hydra/kafka/ingestors/KafkaIngestor.scala (hunk 1 above)
Comment (mcguffin0725) [reply kafka/src/main/scala/hydra/kafka/ingestors/KafkaIngestor.scala]: Check out the KafkaTransport class.  By the time we receive the metadata the ack strategy is gone.  A request has an ack strategy while a response doesn't.

File: kafka/src/main/scala/hydra/kafka/producer/KafkaRecordMetadata.scala [hunk 3]
```diff
@@ -1,15 +1,25 @@
...
-import hydra.core.transport.RecordMetadata
+import hydra.core.transport.{AckStrategy, RecordMetadata}
+import spray.json.{DefaultJsonProtocol, JsNumber, JsObject, JsString, JsValue, RootJsonFormat}
 
 /**
   * Created by alexsilva on 2/22/17.
   */
-case class KafkaRecordMetadata(offset: Long, timestamp: Long, topic: String, partition: Int,
-                               deliveryId: Long) extends RecordMetadata
+case class KafkaRecordMetadata(offset: Long, timestamp: Long, destination: String, partition: Int,
```
Comment (ivanwakeup) [inline kafka/src/main/scala/hydra/kafka/producer/KafkaRecordMetadata.scala]: This is still a KafkaRecord, doesn't it make sense to keep the field name as "topic"? I understand wanting to be decoupled from kafka but this case class isn't

**Comment by roarking** [comment]
It is because the RecordMetadata trait has a destination member variable
that needs to be overriden by subclasses.

**Pull request merged**